Orange/statistics/util.py

```diff
@@ -12,23 +12,8 @@
 from scipy import sparse as sp
 import scipy.stats.stats
 from sklearn.utils.sparsefuncs import mean_variance_axis
-
-
-def sparse_array_equal(x1, x2):
-    """Check if two sparse arrays are equal."""
-    if not sp.issparse(x1):
-        raise TypeError("`x1` must be sparse.")
-    if not sp.issparse(x2):
-        raise TypeError("`x2` must be sparse.")
-
-    return x1.shape == x2.shape and (x1 != x2).nnz == 0
-
-
-def array_equal(x1, x2):
-    """Equivalent of np.array_equal that properly handles sparse matrices."""
-    if sp.issparse(x1) and sp.issparse(x2):
-        return sparse_array_equal(x1, x2)
-    return np.array_equal(x1, x2)
+# For backwards compatibility
+from Orange.data.util import array_equal  # pylint: disable=unused-import
```

Comment:
I don't think we want to have it duplicated here for "backwards compatibility" forever?
Can you wrap this in `Orange.util.deprecated` and make it suggest the use of the preferred `Orange.data.util.array_equal` or something similar. A test which reminds us to remove it in 2 versions time or at least a comment with the timeline for an eventual removal wouldn't hurt either.

Reply:
You're absolutely right. To make sure we remove it, I've added a test that will fail in two versions time i.e. this is included in the 3.22 release and it will fail once we reach 3.24.